### DISCUSSION TITLE
[April 2024] zkSync Developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#514] [Announcements]
Hello developers! Welcome to the April edition of the zkSync Developer Update. Here we showcase the latest advancements in the protocol, a range of new and improved development tools, recent integrations, and a variety of tutorials.

Here are the updates for the last month:

### 🧰 Developer tools

- **hardhat-zksync plugin**: we’ve released a simplified plugin for Hardhat, `hardhat-zksync`. This plugin combines the features from all different plugins we’ve built over the last months (hardhat-zksync-deploy, hardhat-zksync-upgradable, hardhat-zksync-solc…) into a single one for an improved developer experience. Install a single plugin, get all the zkSync-specific Hardhat features. Find [more info in our docs](https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync.html).
- **Viem**: updates in `viem/zksync` to include default fee values in deployment transactions (EIP712). In addition, the extension now includes new paymaster utility methods. More info in [the viem zkSync docs](https://viem.sh/zksync).
- **zksync-ethers SmartAccount class:** the latest release of `zksync-ethers` includes a new account class that simplifies the usage of native account abstraction with pre-configured ECDSA and multisig smart contract accounts. [Learn more in our docs](https://docs.zksync.io/build/sdks/js/zksync-ethers/accounts.html#smartaccount).
- **Remix plugin update:** Nethermind released a new version of the zkSync Remix plugin which improves its stability. Stay tuned for more updates on Remix soon 👀
- **Compiler explorer:** zksolc is now available in [Compiler Explorer](https://godbolt.org/z/KnY7h1qqd), and online tool that allows developers to see the code that is generated by the zkSync custom Solidity compiler, down to the opcodes.

### ⚒️ Foundry

- **foundry-zksync v0.0.2 alpha3**: this release brings huge improvements in developer experience and compatibility. We’ve replaced the custom commands `zkforge` and `zkcast` with Foundry’s default `forge` and `cast`, and included the optional flag `--zksync` to invoke zkSync’s compiler and VM. Read the [full release notes here](https://github.com/matter-labs/foundry-zksync/releases/tag/v0.0.2-alpha.3).
- **Upstream support**: We've begun efforts to include the work we’ve done in [foundry-zksync](https://github.com/matter-labs/foundry-zksync) into the official Foundry repository. Follow [the issue that introduces all the work the team has been doing in our Foundry fork](https://github.com/foundry-rs/foundry/issues/7624) and show your support!

### ⚗️ Alchemy

- One of the most popular infrastructure providers now supports zkSync RPC nodes. [Start building on zkSync with Alchemy infrastructure](https://www.alchemy.com/zksync).

![CleanShot 2024-05-07 at 17 08 30](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/0ea74440-d5e2-401c-a141-68357c348684)
[Source](https://twitter.com/AlchemyPlatform/status/1785703982775812168)

### 🧑🏻‍💻👩‍💻 Chainlink BlockMagic Hackathon

- This online hackathon from April 29th until June 2nd is sponsored by MatterLabs with [$25.000 in bounties](https://chainlinkblockmagic.devpost.com/) for teams building applications that use Chainlink’s data feeds or implement native account abstraction on zkSync. There are also plenty of workshops so make sure to [check the schedule in the website](https://chain.link/hackathon#schedule).

![CleanShot 2024-05-07 at 17 12 20](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/87284b27-0835-49f1-a305-6b492830f182)
[Source](https://twitter.com/chainlink/status/1784944544594702625)

### 🛡️CodeHawks bug bounty

- In collaboration with Cyfrin’s CodeHawks, we launched a bounty program to analyse and find vulnerabilities in a Foundry codebase. Find [the full audit report here](https://www.codehawks.com/report/clvb821kr0001jzdbi6ggixb0). Congratulations to all winners!

![CleanShot 2024-05-07 at 17 07 52](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/f2f987f5-5e2b-4999-ae5c-bb31f9e1d524)
[Source](https://twitter.com/CodeHawks/status/1787437450970771850)

### 🔗 Stay connected

- Follow [@zkSyncDevs](https://twitter.com/zkSyncDevs) on X
- Join the #dev-general channel in [our Discord](https://join.zksync.dev/)

Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by 0xRektor** [comment]
Thanks for the news!

**Comment by King7up** [comment]
Thanks for this zkSync Developer Update, really appreciate your hard work 👍
